data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -0,0 +1,55 @@
+package com.microsoft.azure.kusto.data;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+/*
+ * Kusto supports attaching various properties to client requests (such as queries and control commands).
+ * Such properties may be used to provide additional information to Kusto (for example, for the purpose of correlating client/service interaction),
+ * may affect what limits and policies get applied to the request, and much more.
+ * For a complete list of available client request properties
+ * check out https://docs.microsoft.com/en-us/azure/kusto/api/netfx/request-properties#list-of-clientrequestproperties
+ */
+public class ClientRequestProperties {
```

Comment:
This class generates many access warnings on the public methods, do we need all of them?
for example - the toJson is used internally, isn't it?

Reply:
public is not bad 😄  if you feel any of them should be private, I will change, but client might want to use any of those

Reply:
Why would I (as a developer) would need it in a json format?